manifests/charts/istio-cni/templates/daemonset.yaml

```diff
@@ -92,8 +96,6 @@ spec:
               path: /readyz
               port: 8000
           securityContext:
-            appArmorProfile:
```

Comment:
```
{{- if and (gte .Capabilities.KubeVersion.Major 1) (gte .Capabilities.KubeVersion.Minor 31) }}
            appArmorProfile:
              type: Unconfined
{{- end }}
```

is probably better than removing this.

The annotation can stay (or we can fence it too) and it can be removed when 1.29/1.30 drops off.

Reply:
(if you want to test this with `helm template` you can use the `--kube-version` flag, e.g. `helm template --kube-version 1.31`)

Reply:
One issue is most users doing `template` (which includes argo, etc) aren't going to pass --kube-version, which causes helm to pick a semi-arbitrary version. This is almost certainly <1.31 today, but some random helm upgrade could change it.

Given the annotation works with 1.31+ (?) for now, I would feel a bit more comfortable with just annotation

Reply:
Why not both?

If they template and don't set `kube-version`, they still get the annotation, which works until it doesn't. If they do, they get the non-annotation version if their K8S supports it.

If the templating constraint is actually a k8s version (as it is in this case) it's much more self-documenting to make that explicit in the template - then in 6-12 months we won't have to guess, and anyone looking at the chart can see what the deal is.

Reply:
My worry is purely that KubeVersion is only loosely correlated to the real cluster version due to pervasive usage of `template` (without --kube-version), so if that default becomes "1.31" everyone will break.

I didn't realize you said keep the annotation, though. In that case, it would require helm to default to 1.31 or higher AND the user to be below 1.31 AND use `template` -- maybe very unlikely

Reply:
Anno is pretty easy to fence too in a way that will use it if k8s version unset or verspec out of band:


```
{{- if or (eq .Capabilities.KubeVersion.Version "") (and (ge .Capabilities.KubeVersion.Major "1") (ge .Capabilities.KubeVersion.Minor "31")) }}
        # Add AppArmor annotation
        # This is required to avoid conflicts with AppArmor profiles
        # which block certain privileged pod capabilities.
        container.apparmor.security.beta.kubernetes.io/install-cni: unconfined
{{ end }}
```